blaze-client/src/test/scala/org/http4s/blaze/client/PoolManagerSuite.scala

```diff
@@ -261,4 +263,38 @@ class PoolManagerSuite extends Http4sSuite with AllSyntax {
       _ <- pool.borrow(key)
     } yield assert(conn1.connection.isClosed)
   }
+
+  test("Should not borrow connection if it exceeds the maxBorrowDuration") {
+    for {
+      pool <- mkPool(
+        maxTotal = 1,
+        maxIdleDuration = Duration.Inf,
+        maxBorrowDuration = Duration.Zero,
+      )
+      _ <- pool
+        .borrow(key)
+        .as(false)
+        .recover { case ConnectionBorrowingException(_) =>
+          true
+        }
+        .assert
+    } yield ()
+  }
+
+  test("Should borrow connection if it doesn't exceed the maxBorrowDuration") {
+    for {
+      pool <- mkPool(
+        maxTotal = 1,
+        maxIdleDuration = Duration.Inf,
+        maxBorrowDuration = 2.seconds,
```

Comment:
Anecdotally says, but at local, this test-suite takes about 0.001s on execution. However, at CI it may become flaky. But still, I didn't mark it as 'flaky' before receiving any facts about its flakiness.

Reply:
Lovely. It turned out that the other test suite became flaky. ![mehh](https://pbs.twimg.com/profile_images/1349862091201646593/_uRborDm_400x400.jpg)